simple_messenger/views.py

```diff
@@ -1 +1,136 @@
 ''' views.py '''
+from uuid import UUID
+
+import arrow
+from flask import Blueprint, jsonify, request
+
+from simple_messenger.exceptions import SimpleMessengerException
+from simple_messenger.models import (
+    db,
+    Message,
+    User,
+)
+
+
+api = Blueprint('api', __name__, url_prefix='/api')
+
+
+@api.route('/users', methods=['POST'])
+def new_user():
+    user = User()
+    db.session.add(user)
+    db.session.commit()
+
+    return (
+        jsonify(user.serialize()),
+        201,
+        {'Location': f'/api/users/{user.user_id}'}
+    )
+
+
+@api.route('/users', methods=['GET'])
+def list_users():
+    users = db.session.query(User).all()
+    return jsonify([user.serialize() for user in users]), 200
+
+
+@api.route('/users/<uuid:user_id>', methods=['GET'])
+def get_user(user_id):
+    user = db.session.query(User).get_or_404(user_id)
+    return jsonify(user.serialize()), 200
+
+
+@api.route('/messages', methods=['POST'])
+def new_message():
+
+    message_data = request.json
+    sender = User.get_or_create(message_data['sender_id'])
+    receiver = User.get_or_create(message_data['receiver_id'])
+
+    message = Message(
+        message_text=message_data['message_text'],
+        sender_id=sender.user_id,
+        receiver_id=receiver.user_id,
+    )
+    db.session.add(message)
+    db.session.commit()
+
+    return (
+        jsonify(message.serialize()),
+        201,
+        {'Location': f'/api/messages/{message.message_id}'}
+    )
+
+
+@api.route('/messages/recent', methods=['GET'])
+def list_messages():
```

Comment:
All of the views are pretty simple except this one. I ended up implementing all the various query patterns here (and making them optional). So you can since/limit to your hearts content and also provide params to specify a conversation to filter by. The query logic got kinda hairy here, not sure if this is the best idea...